core/src/main/java/hudson/util/io/TarArchiver.java

```diff
@@ -59,7 +59,7 @@
     public void visitSymlink(File link, String target, String relativePath) throws IOException {
         TarArchiveEntry e = new TarArchiveEntry(relativePath, TarConstants.LF_SYMLINK);
         try {
-            int mode = IOUtils.mode(link);
```

Comment:
My only explanation for you comment is that you might have missed added line following that line.